### PR TITLE
{bio}[foss/2017b,intel/2017b] GROMACS v2018.2

### PR DESCRIPTION
**Pull request by vanzod**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/0cde96e6bf7f532a66cba9714066d626 for a full test report.

**Comment by vanzod** [comment]
@migueldiascosta I presume the tests failures with the Intel compiler are due to the issue described in this bug:
https://redmine.gromacs.org/issues/2335

That should have been fixed almost a year ago though...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2641.swalot.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5ec4d22716129e8679210990df9bda6e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2063.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/13a57273c0d8c42acf47f8bb7daf1369 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node3147.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/8c7b54f0ef5c03883651cc018de4c2bb for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 11 out of 11 (2 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/ebe4f70862b245ed7a31d412d44186a8 for a full test report.

**Comment by akesandgren** [comment]
@migueldiascosta @boegel can you check the complete debug log of your failed builds and dig out the exact compiler command with flags and all?

**Comment by akesandgren** [comment]
Also found a "bug" with at least the intel easyconfig here, it lacks hwloc as dependency.
See the configure part of the log, at least mine says:
```
-- hwloc version: 1.11.5
-- Found Hwloc: /usr/lib/x86_64-linux-gnu/libhwloc.a (found suitable version "1.1
1.5", minimum required is "1.5") 
-- Static hwloc library found, will not attempt using it as it could lead to link
-time errors. To use the detected library, manually set GMX_HWLOC=ON and you will
 likely have to pass appropriate linker flags too to satisfy the link-time depend
encies of your hwloc library. Try "pkg-config --libs --static hwloc" for suggesti
ons on what you will need.
```

**Comment by akesandgren** [comment]
And reading the Gromacs issue, it says that:
"ICC enables flush-to-zero for denormals by default, which will cause errors for very small arguments without FMA support since some intermediate results might be denormal."
At least my build explicitly turns on -ftz, which means that SandyBridge which don't have FMA (if i remember correctly) will see problems.

**Comment by vanzod** [comment]
@akesandgren I confirm that `hwloc` is missing and should be added to the `GROMACS` dependencies:
```
-- Could NOT find Hwloc (missing: HWLOC_LIBRARIES HWLOC_INCLUDE_DIRS) (Required is at least version "1.5")
```

**Comment by vanzod** [comment]
@akesandgren So, using `'strict': True` as toolchain option removes the `-ftz` and all tests pass on Sandy Bridge. What is the downside of using this option for all other architectures?

**Comment by akesandgren** [comment]
From a correctness perspective it is good, from a speed perspective it might be bad.
The only way to know is to test.
But personally I would rather be able to turn on/off ftz irrespectively of the other precision flags.
It would require a rather small change to toolchains/compiler/inteliccifort.py but one would have to do it carefully to not change current behaviour.

**Comment by vanzod** [comment]
@akesandgren I tested with `'loose': True` but it fails again. So the only way out is to use `'strict': True`.

**Comment by vanzod** [comment]
@akesandgren Good news. `'precise': True` also works. Thoughts about adding it to the easyconfig file?

**Comment by akesandgren** [comment]
I'd rather have a comment in it mentioning that it will be needed for cpu's that don't have FMA.
It is, nowadays, an exception to have old cpus that don't have FMA.

Or change the intel options to be able to turn on/off ftz explicitly outside of precflags.

**Comment by boegel** [comment]
Or enhance the existing `GROMACS` easyblock to enable `precise` when running on a system that doesn't support `FMA`?
We have everything in place in framework for implementing that in the easyblock imho...
See https://easybuild.readthedocs.io/en/latest/api/easybuild.tools.systemtools.html#easybuild.tools.systemtools.get_cpu_features (see the `FFTW` easyblock how this can be leveraged).

**Comment by akesandgren** [comment]
Ah yes.... didn't think of that.
But regardless, being able to turn ftz on/off on demand is a good thing to be able to do.
I'll try to remember to make a issue or PR about this...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3131.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/e9993d34ef7081dfba1e4e6b6770b18c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2709.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/97a1c115b9171682157586d2fe9b7ce5 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2019.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/fb230dd1311295736d82733aed0264b6 for a full test report.

**Comment by boegel** [comment]
re-tested on top of https://github.com/easybuilders/easybuild-easyblocks/pull/1706

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2018.2-foss-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+GROMACS is a versatile package to perform molecular dynamics,
+ i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles.
+
+This is a CPU only build, containing both MPI and threadMPI builds.
+"""
+
+toolchain = {'name': 'foss', 'version': '2017b'}
+toolchainopts = {'openmp': True, 'usempi': True}
+
+source_urls = ['ftp://ftp.gromacs.org/pub/gromacs/']
```
Comment (boegel) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2018.2-foss-2017b.eb]: @vanzod Please prefer `https://` over `ftp://`, cfr. #7948

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2018.2-intel-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+GROMACS is a versatile package to perform molecular dynamics,
+ i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles.
+
+This is a CPU only build, containing both MPI and threadMPI builds.
+"""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+toolchainopts = {'openmp': True, 'usempi': True}
+
+source_urls = ['ftp://ftp.gromacs.org/pub/gromacs/']
```
Comment (boegel) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2018.2-intel-2017b.eb]: @vanzod Please prefer `https://` over `ftp://`, cfr. #7948

**Comment by boegel** [comment]
Going in, thanks @vanzod!

**Pull request merged**